mdata/store_cassandra.go

```diff
@@ -204,9 +204,16 @@ func (c *cassandraStore) processWriteQueue(queue chan *ChunkWriteRequest, meter
 
 			data := cwr.chunk.Series.Bytes()
 			chunkSizeAtSave.Value(int64(len(data)))
-			version := chunk.FormatStandardGoTsz
+			version := chunk.FormatStandardGoTszWithSpan
 			buf := new(bytes.Buffer)
-			binary.Write(buf, binary.LittleEndian, uint8(version))
+			binary.Write(buf, binary.LittleEndian, chunk.Format(version))
+
+			chunkSize, ok := chunk.RevChunkSpans[cwr.span]
+			if !ok {
+				// it's probably better to panic than to persist the chunk with a wrong length
+				panic(fmt.Sprintf("Chunk size invalid: %d", cwr.span))
```

Comment:
I don't pretend to grok how this fits into the larger scheme of things, and if it's a scenario where the best course of action is to shut down MT altogether then that's fine, I just want to be sure it's not a scenario where we end up throwing the baby out with the bathwater.